lms/djangoapps/support/views/contact_us.py

```diff
@@ -33,10 +37,34 @@ def get(self, request):
 
         if request.user.is_authenticated():
             context['user_enrollments'] = CourseEnrollment.enrollments_for_user_with_overviews_preload(request.user)
+            last_accessed_course = self.get_last_accessed_course(request, context['user_enrollments'])
+            context['course_id'] = last_accessed_course.get('course_id')
+
             enterprise_learner_data = enterprise_api.get_enterprise_learner_data(user=request.user)
             if enterprise_learner_data:
                 tags.append('enterprise_learner')
 
         context['tags'] = tags
 
         return render_to_response("support/contact_us.html", context)
+
+    def get_last_accessed_course(self, request, enrollments):
+        """Get learner's last accessed course."""
+        dates = []
+        for enrollment in enrollments:
+            course = get_course_by_id(enrollment.course_overview.id)
+            field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
```

Comment:
`FieldDataCache` does not add cache value until you access any block/chapter of the course. It does not work if you only access the parent course page. e.g https://single-support-form.sandbox.edx.org/courses/course-v1:edX+DemoX+Demo_Course/course/

Reply:
@tasawernawaz Yes, this implementation has this drawback. I think we can move back to the previous one which is more accurate.